### PR TITLE
Elastic training

### PR DESCRIPTION
**Pull request by robertjrodger**
This PR creates a subdirectory called `elastic/` in the root directory, which contains a version of the already-existing `nlp/` subdirectory, modified for Elastic Stack trainings (Elasticsearch, Logstash, Kibana, Filebeat).

### CODE REVIEW

File: elastic/deploy-elastic.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+def get_instance_name_prefix(name, N_sequence=10):
+    if name:
+        return f"elastic-{name}"
+    else:
+        return f"elastic-{get_random_id(N_sequence)}"
+
+
+def create_machines(project_id, name_prefix, instances):
+    # to_run = f"""gcloud dataproc --region europe-west1 clusters create {cluster_name}
```
Comment (krisgeus) [inline elastic/deploy-elastic.py]: Can you remove the comments

File: elastic/deploy-elastic.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    # """
+    for instance in range(0,instances):
+    	machine_name = f"{name_prefix}-{instance}"
+    	to_run = f"""gcloud beta compute instances create {machine_name} 
+    	--subnet=default --zone=europe-west1-c
+    	--machine-type=n1-standard-4
+    	--tags elastic-training-instance
+    	--boot-disk-size=20GB --boot-disk-type=pd-standard --boot-disk-device-name={machine_name}
+		"""
+    	# --network-tier=PREMIUM
```
Comment (krisgeus) [inline elastic/deploy-elastic.py]: Can you remove the comments

File: elastic/deploy-elastic.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    	# --service-account=787238606102-compute@developer.gserviceaccount.com
+    	# --scopes=https://www.googleapis.com/auth/devstorage.read_only,https://www.googleapis.com/auth/logging.write,https://www.googleapis.com/auth/monitoring.write,https://www.googleapis.com/auth/servicecontrol,https://www.googleapis.com/auth/service.management.readonly,https://www.googleapis.com/auth/trace.append
+    	# --tags=http-server,https-server
+    	# --image=debian-9-stretch-v20180510
+    	# --image-project=debian-cloud 
+    	if project_id:
+        	to_run += f" --project_id {project_id}"
+    	c = delegator.run(to_run.replace('\n', ' '))
+    	if c.err:
+			# don't raise, it's printing to standard error but it's not an error
```
Comment (krisgeus) [inline elastic/deploy-elastic.py]: Too much white space

**Comment by robertjrodger** [comment]
@krisgeus irrelevant comments removed and whitespace adjusted.

i agree that there's a lot of overlap between `nlp/` and `elastic`. I think that if lots of projects plan to use this, maybe it makes sense have one base master branch and then each training has a separate branch that transforms the base branch for their purposes.

**Pull request merged**